crates/client/src/rr/tsig.rs

```diff
@@ -0,0 +1,354 @@
+// Copyright 2015-2019 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! tsigner is a structure for computing tsig messasignuthentication code for dns transactions
+use crate::proto::error::{ProtoError, ProtoResult};
+use crate::proto::rr::rdata::tsig::{
+    make_tsig_record, message_tbs, signed_bitmessage_to_buf, Algorithm, TSIG,
+};
+use std::ops::Range;
+use std::sync::Arc;
+
+use crate::op::{DnsResponse, Message, MessageFinalizer, MessageVerifier};
+use crate::rr::{Name, RData, Record};
+
+/// Struct to pass to a client for it to authenticate requests using TSIG.
+#[derive(Clone)]
+pub struct TSigner(Arc<TSignerInner>);
```

Comment:
Does clippy warn on this name in current rust versions? I think it might suggest `TSigner` be `Tsigner`.

Reply:
CI passed so I guess it's fine for clippy. I'm not sure how the lint for acronyms works, but it seems to allow quiet a few upper case letters as long as some are lowercase (the example in https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms does not even appear to actually print warning for me)